src/entities/fractions/fraction.rs

```diff
@@ -57,21 +64,23 @@ where
     }
 
     // Returns the remainder after floor division as a new fraction
-    fn remainder(&self) -> Self {
+    fn remainder(&self) -> Result<Self, Error> {
```

Comment:
`FractionTrait::new` only fails when denominator is zero, which is not possible here.